### DISCUSSION TITLE
All objects is going out when run on Web Assembly.

### DISCUSSION THREAD

**Discussion by paulocoutinhox** [axmolengine/axmol#2041] [General]
Hi,

When i build my game for Web Assembly (wasm) i have a strange problem or bug (i don't know), that all objects simple disappear.

See this video:

https://github.com/user-attachments/assets/65c71a7b-a36a-4017-9e87-725967534f4f

Can anyone help me understand why?

Thanks.

**Comment by Tuan-Git-Hub** [comment]
Have you resolved this issue yet? I'm facing a similar problem in my project.

**Reply by paulocoutinhox** [reply]
I dont solved it, still the same problem on WASM.

**Reply by aismann** [reply]
Which Axmol version ?

**Reply by Tuan-Git-Hub** [reply]
I'm using axmol version 2.4, and the problem only occurs when building on wasm, building on a PC or Android works perfectly fine.

**Comment by halx99** [comment]
It's strange, could you build with ImGui inspector to check does your objects in scene tree?

**Reply by Tuan-Git-Hub** [reply]
I haven't used the ImGui inspector, but these objects can still be seen in the scene tree, but like in video, they seem to flash in the next frame. PC and Android builds are normal. Anyway, this is just a learning project, so in other projects I'll check more carefully from the start when building on wasm.

**Reply by aismann** [reply]
> . Anyway, this is just a learning project, ...

Is it public and the source can be checked?

**Reply by Tuan-Git-Hub** [reply]
Yes, you can view the source code here: https://github.com/Tuan-Git-Hub/Game_Frogventure

**Comment by halx99** [comment]
@paulocoutinhox do you use axmol 2.x pysics2d based on chipmunk ?

**Reply by paulocoutinhox** [reply]
Yes.

**Reply by halx99** [reply]
The physics2d module in Axmol 2.x, which is based on Chipmunk, behaves abnormally under WebAssembly. This seems to be an issue with Chipmunk itself, and there are currently no plans to conduct an in‑depth investigation into the Chipmunk library.
